File: Data Structures & Algorithms/two-integer-sum/submission-7.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        res_idx = {}
        for i, num in enumerate(nums):
            if num in res_idx:
                return [res_idx[num], i]
            
            res_idx[target - num] = i

        return [-1, -1]



# [3,4,5,6]

# res_idx = {
#     4: 0

# }

File: Data Structures & Algorithms/top-k-elements-in-list/submission-7.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> m;
        for (auto& num : nums) {
            m[num]++;
        }

        priority_queue<pair<int, int>, vector<pair<int, int>>, greater<pair<int, int>>> pq;
        for (auto& [num, freq] : m) {
            pq.push(pair<int, int>{freq, num});
            if (pq.size() > k) {
                pq.pop();
            }
        }

        vector<int> result;
        while (!pq.empty()) {
            result.push_back(pq.top().second);
            pq.pop();
        }

        return result;
    }
};
